### DISCUSSION TITLE
autorization issue

### DISCUSSION THREAD

**Discussion by IraSfx** [jgraph/drawio#4257] [General]
### Preflight Checklist

* [ ] I agree to follow the [Code of Conduct](https://github.com/jgraph/drawio/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [ ] I have searched the issue tracker for a feature request that matches the one I want to file, without success.

**Describe the bug**

I have the google doc add-on and I'm trying to add an existing diagram, I do have the option to create a new one and to update an existing one but not insert. 

This is the error message that I have received

Error: ScriptError: Authorization is required to perform that action.

I have tried to uninstall and install again the add-on
add the diagram
![Screen Shot 2020-04-22 at 23 50 40](https://user-images.githubusercontent.com/64167496/80032650-28d29f00-84f4-11ea-9197-483a433a5503.png)
 from the web app to the doc.

**Screenshots**
attached
**draw.io version (In the Help->About menu of the draw.io editor):**

- 13.0.1

**Desktop (please complete the following information):**

**Comment by davidjgraph** [comment]
Try with those browser extensions switched off.

**Comment by dcsmconsultancy** [comment]
It seems many users are having the exact same issue and is resolved by using a different browser or new ingognito window for cases where you have multiple google accounts and you are not using the default account.

Is there a way for the script to look up the account from the active tab?  This would solve the issues.  Otherwise we have to then either open a private window or save an image and insert that.

**Comment by theprojectsomething** [comment]
Can confirm this error with two google accounts, specifically where the default account does not have the add-on installed. This error shows when attempting to insert a diagram into a doc created by the secondary account.

On installing for both accounts, only the default drive is accessible via the add-on menu (regardless of the active account).

**Comment by calebeno** [comment]
Wanted to follow up on this.  I am also having this issue and would love to see it fixed.  What needs to be done to make progress?

**Comment by gezamolnar** [comment]
Hi,
It is still an issue.
Annoying one.

The tool is good - I hope it's going to be fixed very soon, after a year.

Cheers,
   Géza

**Comment by migscabral** [comment]
I'm also experiencing the same issue when I have multiple Google Accounts logged in, and not using the main account.

**Comment by celiasmith** [comment]
This remains a critical issue for me still.

**Comment by KaleabTessera** [comment]
Still an important issue.

**Comment by mamacdon** [comment]
Not completed? I still see this bug in v3.7 of the extension

![image](https://user-images.githubusercontent.com/382404/201945585-42a060be-d5f6-451c-8e14-4ebf3dad2cc0.png)

**Comment by mehrbat** [comment]
I have multiple Google accounts, and the default one (what does that mean though?) does not have it installed. 
I have this issue happening to the docs created with my new G-Suite account.

**Comment by m-mohamedin** [comment]
Unfortunately, we are limited by Google APIs and so far we cannot fix this from our side

**Comment by jamesharr** [comment]
Confirming that this still seems to be an issue.

The steps to reproduce are pretty straight forward:
- In an incognito tab
- Login to Google Account 1 (one where you do NOT intend to use draw.io)
- Login to Google Account 2 (one where you DO intend to use draw.io)
- Create a new Google Doc in Google Account 2
- Attempt to insert diagram

The work-around is also straight forward, although inconvenient:
- Use a separate browser instance, an incognito/private tab (most browsers), or a Container Tab (Firefox)
- Login to the google account you want to use (and only this account)
- Continue using the draw.io for google docs to insert the diagram

Note that based on my experience, the work-around is only required to insert diagrams. It is not required to view, edit, or update diagrams. Please correct me if I am wrong.

**Comment by jamesharr** [comment]
If the software team is unable to fix the issue due to an upstream issue, then it might be worth providing the user some hints/feedback as to a work-around (either a link to an FAQ, or some embedded help)

**Comment by toLucky81** [comment]
I can see that the last entry was 1½ years ago. But, the issue isnt resolved. 

Im logged in with:
1) my google doc on account B
2) my draw.io drawing on account B
3) google drive on account A

Is anyone working to resolve this?
Thanks

**Comment by juicejuice** [comment]
Yes this is still occurring. I have a personal Google account logged in and a work Google account. I'm not sure how it knows what the default is, but I do need both logged in. For other Google apps it will prompt which account to use, but draw.io doesn't seem to do this so I'm guessing it picks the wrong account. This is a pretty major limitation for my use case.
EDIT: if I add the draw.io for Google Docs add-on to my personal Google account then the error goes away but I can only access my personal files, not my work files. This makes the add-on useless.

**Reply by davidjgraph** [reply]
You're saying that you have google integrated, third-party apps , that are able to detect the account correctly in their doc add-in integrations? Could you name those please?

**Reply by juicejuice** [reply]
@davidjgraph please accept my apologies, I think my message was unclear. I meant for first-party Google Apps it works to have two accounts as it will prompt for which account to use. I don't use any other add-ons except for draw.io as it's the best drawing tool. I understand this issue is a Google limitation. My workaround for now is to PNG my diagrams and insert them that way ;(
